backend/src/apiserver/template/template_test.go

```diff
@@ -73,7 +73,7 @@ kind: Workflow`,
 		templateType: V1,
 	}, {
 		template:     v2SpecHelloWorldJSON,
-		templateType: V2,
+		templateType: Unknown,
```

Comment:
With this change, we lost the coverage for when templateType == V2. We should definitely have a positive case for that.

Reply:
The negative (json) test case can be short and inlined here, doesn't need to be the full valid json IR.

Reply:
> With this change, we lost the coverage for when templateType == V2. We should definitely have a positive case for that.

The test file includes another test case with V2 template in yaml. So I think it is covered?



> The negative (json) test case can be short and inlined here, doesn't need to be the full valid json IR.

There is also another test case in JSON format, the content of which is not pipeline spec though. Maybe I can just remove the JSON test case (with entire pipeline)?